rclpy/rclpy/__init__.py

```diff
@@ -12,12 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import rclpy._rclpy__rmw_opensplice_cpp as _rclpy
+import importlib
+import os
+
+import ament_index_python
 
 from rclpy.node import Node
 
+rmw_implementations = sorted(ament_index_python.get_resources('rmw_implementation').keys())
+
+
+def _load_rmw(rmw_implementation):
+    module_name = '._rclpy__{rmw_implementation}'.format(
+        rmw_implementation=rmw_implementation,
+    )
+    return importlib.import_module(module_name, package='rclpy')
+
+
+global _rclpy
+try:
+    _rclpy
+except NameError:
+    _rclpy = None
+if _rclpy is None:
+    for rmw_implementation in rmw_implementations:
+        rmw_module = _load_rmw(rmw_implementation)
+        if rmw_module is not None:
+            _rclpy = rmw_module
+            break
+
+
+def set_rmw_implementation(rmw_implementation):
```

Comment:
Hmm, I think this function can't be allowed to be used at any point during run time. I believe loading more than one of the rmw implementations simultaneously will end badly (the libraries export the same symbols). For this reason, I think any selection of which rmw implementation to use must be done before importing `rclpy` and can only be done once.

I think you'll have to use a singleton "selector" module that must be used before `import rclpy`, or you'll have to use some external setting like an environment variable or a configuration file.

This is sort of what I was thinking of:

```python
# User's code
import rclpy_selector
rclpy_selector.select('name of rmw implementation')
import rclpy
rclpy_selector.select('another name')  # Raises an exception that says selection cannot happen after importing rclpy
```

And `rclpy` would look something like this:

```python
# within rclpy.py
import rclpy_selector
_rclpy = rclpy_selector.import_rmw_implementation()
```

I'm not sure that will work, but it's just what I had imagined.